proto/osmosis/validator-preference/v1beta1/tx.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package osmosis.validatorpreference.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+import "osmosis/validator-preference/v1beta1/state.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v12/x/validator-preference/types";
+
+// Msg defines the validator-preference modules's gRPC message service.
+service Msg {
+  // CreateValidatorSetPreference creates a set of validator preference.
+  rpc CreateValidatorSetPreference(MsgCreateValidatorSetPreference)
+      returns (MsgCreateValidatorSetPreferenceResponse);
+
+  // UpdateValidatorSetPreference updates an existing set of validator
+  // preference.
+  rpc UpdateValidatorSetPreference(MsgUpdateValidatorSetPreference)
+      returns (MsgUpdateValidatorSetPreferenceResponse);
+
+  // StakeToValidatorSet gets the owner, coins and stakes to a validator-set.
+  rpc StakeToValidatorSet(MsgStakeToValidatorSet)
+      returns (MsgStakeToValidatorSetResponse);
+
+  // UnStakeFromValidatorSet gets the owner and coins and unstakes from
+  // validator-set. The unbonding logic will follow the `Undelegate` logic from
+  // the sdk.
+  rpc UnStakeFromValidatorSet(MsgUnStakeFromValidatorSet)
+      returns (MsgUnStakeFromValidatorSetResponse);
+
+  // WithdrawDelegationRewards allows users to claim rewards from the
+  // validator-set.
+  rpc WithdrawDelegationRewards(MsgWithdrawDelegationRewards)
+      returns (MsgWithdrawDelegationRewardsResponse);
+}
+
+// MsgCreateValidatorSetPreference is a list that holds validator-set.
+message MsgCreateValidatorSetPreference {
```

Comment:
```suggestion
message MsgSetValidatorSetPreference {
```

Reply:
I think we just make 1 message, `SetValidatorSetPreference` wdyt?

Reply:
I agree with @ValarDragon here